src/hotspot/share/runtime/thread.cpp

```diff
@@ -1750,13 +1761,13 @@ void JavaThread::send_thread_stop(oop java_throwable)  {
 //   - Target thread will not enter any new monitors.
 //
 bool JavaThread::java_suspend() {
-  guarantee(Thread::is_JavaThread_protected(this, /* checkTLHOnly */ true),
+  guarantee(Thread::is_JavaThread_protected_by_TLH(/* target */ this),
```

Comment:
Nit: there's no need to comment a single argument.

Reply:
I added the comment due to this comment from @coleenp:
```
At the callsites below, it would make it obvious that "this" isn't meant to be the current thread (and calling Thread::current() inside a (tbd) assert would make it clear which thread is which.
```

I read @coleenp's comment as meaning that the uncommented `this` could be made more
obvious so I'm going to leave those two `/* target */` comments for clarity.